### ISSUE TITLE
Crash in TextInput SetDimensions after upgrading from ReactNativeWindows 0.50 to 0.53

### ISSUE THREAD

**Issue by apparel21ci** [Microsoft/react-native-windows#1673]
### Environment

1. `react-native -v`:
react-native-cli: 2.0.1
react-native: 0.53.3
2. `npm ls rnpm-plugin-windows`: rnpm-plugin-windows@0.2.8
3. `npm ls react-native-windows`: react-native-windows@0.53.0-rc.1
4. `node -v`: v8.1.2
5. `npm -v`: 5.5.1
6. `yarn --version`: 1.3.2

- Target Platform: UWP
- Target Platform Version(s): 10.0.14393
- Target Device(s): Desktop
- Development Operating System: Windows 10
- Visual Studio Version: 2017 version 15.5.7

### Steps to Reproduce

I am attempting to reproduce a simple example, but it is difficult. The TextInput in our application which is having this problem is buried deep in lots of other screens. I am trying to isolate the causing conditions, but not able to yet.
  From my investigations, the NaN value is only passed on initialisation. If I set a conditional breakpoint on the line 
view.MinWidth = dimensions.Width;
and set the dimensions.Width & Height to 1, it never hits the breakpoint again even if resizing, minimising etc. 

### Expected Behavior

Not crash 

### Actual Behavior

An exception is raised in the Windows code which crashes the application. 
![image](https://user-images.githubusercontent.com/8953785/36702593-dd422abe-1bab-11e8-8f47-85141be1ce54.png)

```
   at Windows.UI.Xaml.FrameworkElement.put_MinWidth(Double value)
   at ReactNative.Views.TextInput.ReactTextInputManager.SetDimensions(ReactTextBox view, Dimensions dimensions)
   at ReactNative.UIManager.ViewManager`2.ReactNative.UIManager.IViewManager.SetDimensions(DependencyObject view, Dimensions dimensions)
   at ReactNative.UIManager.NativeViewHierarchyManager.UpdateLayout(DependencyObject viewToUpdate, IViewManager viewManager, Dimensions dimensions)
   at ReactNative.UIManager.NativeViewHierarchyManager.UpdateLayout(Int32 parentTag, Int32 tag, Dimensions dimensions)
   at ReactNative.UIManager.UIViewOperationQueue.<>c__DisplayClass25_0.<EnqueueUpdateLayout>b__0()
   at ReactNative.UIManager.UIViewOperationQueue.<>c__DisplayClass34_1.<DispatchViewUpdates>b__1()
   at ReactNative.UIManager.UIViewOperationQueue.OnRendering(Object sender, FrameEventArgs e)
   at ReactNative.UIManager.UIViewOperationQueue.OnRenderingSafe(Object sender, FrameEventArgs e)

```
![image](https://user-images.githubusercontent.com/8953785/36702675-2d58867e-1bac-11e8-9a8d-cb89b34e7d0e.png)

**Comment by rozele** [comment]
It's concerning that Yoga is sending a NaN value as opposed to 0. This speaks to a bigger issue with the layout engine. Would it be possible to experiment with the master branch, where we've integrated the latest from https://github.com/facebook/yoga to see if that has any impact on the issue?

**Comment by jmcginty** [comment]
Sure, I can give that a try

**Comment by jmcginty** [comment]
Having trouble building with the master branch. I changed my package.json to refer to 
from		"react-native-windows": "0.53.0-rc.1",
to		"react-native-windows": "Microsoft/react-native-windows",

However when running the project it complains it can't find the yoga.dll, not sure if I'm going about this the right way. 
```

{System.DllNotFoundException: Unable to load DLL 'yoga': The specified module could not be found. (Exception from HRESULT: 0x8007007E)
   at Facebook.Yoga.Native.YGConfigGetDefault()
   at Facebook.Yoga.YGConfigHandle..cctor()}
```

**Comment by jmcginty** [comment]
@rozele I'd like to try this out in 0.54, There are apparently ["massive"](https://github.com/facebook/react-native/releases/tag/v0.54.0) amounts of yoga optimisations, cleanups and test fixes.
Can we do a 0.54 rc ?

**Comment by rozele** [comment]
@jmcginty 0.54.0-rc.1 is available.

**Comment by jmcginty** [comment]
ripper. Will try again soon.

**Comment by jmcginty** [comment]
Still a problem in 0.54.0-rc.1

**Comment by rozele** [comment]
And... assuming this is still a problem in 0.57 :-)?

**Comment by jmcginty** [comment]
Actually, I think this is fixed in 0.57, so you can close it.

**Comment by rozele** [comment]
Yay!
